drivers/flash/flash_mcux_xspi.c

```diff
@@ -0,0 +1,490 @@
+	flashXfer.seqIndex        = enableOctal ? FLASH_CMD_WRITE_ENABLE_OPI : FLASH_CMD_WRITE_ENABLE;
+
+	return memc_mcux_xspi_transfer(xspi_dev, &flashXfer);
+}
+
+static int flash_mcux_xspi_write(const struct device *dev, off_t offset, const void *data, size_t len)
+{
+	struct flash_mcux_xspi_data *devData = dev->data;
+	const struct device *xspi_dev = devData->xspi_dev;
+	xspi_transfer_t flashXfer;
+	uint32_t key = 0;
+	int ret;
+
+	if (memc_xspi_is_running_xip(xspi_dev)) {
+		key = irq_lock();
+		memc_xspi_wait_bus_idle(xspi_dev);
+	}
+
+	ret = flash_mcux_xspi_write_enable(dev, 0, true);
+	if (ret < 0) {
+		return ret;
+	}
+
+	flashXfer.deviceAddress   = devData->amba_address + offset;
+	flashXfer.cmdType         = kXSPI_Write;
+	flashXfer.seqIndex        = FLASH_CMD_PAGEPROGRAM_OCTAL;
+	flashXfer.targetGroup     = kXSPI_TargetGroup0;
+	flashXfer.data            = (uint32_t *)data;
+	flashXfer.dataSize        = len;
+	flashXfer.lockArbitration = false;
+
+	ret = memc_mcux_xspi_transfer(xspi_dev, &flashXfer);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = flash_xspi_nor_wait_bus_busy(dev, true);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (memc_xspi_is_running_xip(xspi_dev)) {
+		irq_unlock(key);
+	}
+
+	return 0;
+}
+
+static int flash_mcux_xspi_erase_sector(const struct device *dev, off_t offset)
+{
+	struct flash_mcux_xspi_data *data = dev->data;
+	const struct device *xspi_dev = data->xspi_dev;
+	xspi_transfer_t flashXfer = {0};
+
+	flashXfer.deviceAddress   = data->amba_address + offset;
+	flashXfer.cmdType         = kXSPI_Command;
+	flashXfer.seqIndex        = FLASH_CMD_ERASE_SECTOR;
+	flashXfer.targetGroup     = kXSPI_TargetGroup0;
+	flashXfer.lockArbitration = false;
+	flashXfer.dataSize        = 0UL;
+	flashXfer.data            = NULL;
+
+	return memc_mcux_xspi_transfer(xspi_dev, &flashXfer);
+}
+
+static int flash_mcux_xspi_erase(const struct device *dev, off_t offset, size_t size)
+{
+	struct flash_mcux_xspi_data *data = dev->data;
+	const struct device *xspi_dev = data->xspi_dev;
+	uint32_t key = 0;
+	int ret;
+
+	if (offset % SPI_NOR_SECTOR_SIZE) {
+		LOG_ERR("Invalid offset");
+		return -EINVAL;
+	}
+
+	if (size % SPI_NOR_SECTOR_SIZE) {
+		LOG_ERR("Invalid size");
+		return -EINVAL;
+	}
+
+	if (memc_xspi_is_running_xip(xspi_dev)) {
+		key = irq_lock();
+		memc_xspi_wait_bus_idle(xspi_dev);
+	}
+
+	for (int i = 0; i < size / SPI_NOR_SECTOR_SIZE; i++) {
+		ret = flash_mcux_xspi_write_enable(dev, 0, true);
+		if (ret < 0) {
+			return ret;
+		}
+
+		ret = flash_mcux_xspi_erase_sector(dev, offset + i * SPI_NOR_SECTOR_SIZE);
+		if (ret < 0) {
+			return ret;
+		}
+
+		ret = flash_xspi_nor_wait_bus_busy(dev, true);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	if (memc_xspi_is_running_xip(xspi_dev)) {
+		irq_unlock(key);
+	}
+
+	return 0;
+}
+
+static int flash_mcux_xspi_enable_quad(const struct device *dev)
+{
+	struct flash_mcux_xspi_data *data = dev->data;
+	const struct device *xspi_dev = data->xspi_dev;
+	uint32_t value = 0x02;
```

Comment:
can we use some macro or some definition? I guess this value is a standard value. Using meaningful definition is better.